platalea/utils/preprocessing.py

```diff
@@ -17,13 +21,38 @@
 import torchvision.transforms as transforms
 from platalea.experiments.config import get_argument_parser
 
-
 args = get_argument_parser()
 _audio_feat_config = dict(type='mfcc', delta=True, alpha=0.97, n_filters=40,
                           window_size=0.025, frame_shift=0.010)
 _images_feat_config = dict(model='resnet')
 
 
+def extract_audio_from_videos(video_dir_path, audio_dir_path):
+    os.makedirs(audio_dir_path, exist_ok=True)
+    for video_file_name in os.listdir(video_dir_path):
+        video = VideoFileClip(str(video_dir_path / video_file_name))
+        audio_file_name = video_file_name + '.wav'
+        video.audio.write_audiofile(audio_dir_path / audio_file_name)
+        video.close()
+
+
+def preprocess_howto100m(dataset_path, audio_subdir, video_subdir):
+    audio_path = pathlib.Path(dataset_path) / audio_subdir
+    video_path = pathlib.Path(dataset_path) / video_subdir
+    if not audio_path.exists():
+        extract_audio_from_videos(video_path, audio_path)
+
+    extract_howto100m_audio_features(dataset_path, audio_subdir, _audio_feat_config)
+
+
+def extract_howto100m_audio_features(dataset_path, audio_subdir, feat_config):
+    audio_dir_path = dataset_path / audio_subdir
+    file_names = os.listdir(audio_dir_path)
+    paths = [audio_dir_path / fn for fn in file_names]
+    features = audio_features(paths, feat_config)
+    torch.save(dict(features=features, filenames=file_names), dataset_path / 'mfcc_features.pt')
```

Comment:
aren't we supposed to save that as a numpy memmap?

Reply:
good point of course